packages/shared/services/types.ts

```diff
@@ -20,6 +20,12 @@ export type Auth2faType = 'otp' | 'off' | 'optional' | 'on' | 'webauthn';
 
 export type AuthType = 'local' | 'github' | 'oidc' | 'saml';
 
+// PrimaryAuthType defines types where if:
+//  - local: preferred authn is with username and password
+//  - pwdless: preferred authn is passwordless
+//  - sso: preferred authn is either with github, oidc or saml provider
+export type PrimaryAuthType = 'local' | 'pwdless' | 'sso';
```

Comment:
In some places we use full word `passwordless` and in others `pwdless`. Can we stick to one version (I prefer full words).